Sovereign-Labs/sovereign-sdk
# fix keypair/private_key discrepancy for sov-cli

### Description

This PR aims to fix a bug related to the `sov-cli rpc submit-batch` command. The bug causes an error message: `Error: missing field private_key at line 1 column 129`. 

The issue seems to be related to the recent ed25519 upgrade, where the serialized JSON file only contains the `key_pair` and not the `private_key` and `address`.

### Changes

- changed existing key formats to all consistently use the `sov_cli::wallet_state::PrivateKeyAndAddress` format
- changed relevant tests
- changed some other keyfiles to also use the new format
- ensured the new keyfiles private keys are consistent with the bech addresses in the files
- changed README instructions
- modified demo-rollup Makefile to reference the correct key when submitting transactions


### How to Test

1. Run the demo-rollup workflow with `make clean`, `make start`, `make test-create-token` from the demo-rollup folder and ensure that a new token is created as per the README
2. Ensure that other functionalities related to key management are not affected by this change.

---

### Fixes

Fixes #810
